interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -344,8 +344,452 @@ static void StreamValue(llvm::raw_ostream& o, const void* V,
+  std::string printValue(char val) {
+    std::ostringstream strm;
+    if (val > 0x1F && val < 0x7F) strm << "'" << val << "'";
+    else strm << "0x" << std::hex << (int) val << std::dec;
+    return strm.str();
+  }
+
+  std::string printValue(signed char val) {
+    return printValue((char) val);
+  }
+
+  std::string printValue(const unsigned char val) {
+    return printValue((char) val);
+  }
+
+  // Ints
+  std::string printValue(short val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned short val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(int val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned int val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned long long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(float val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(double val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long double val) {
+    std::ostringstream strm;
+    strm << val << "L";
+    return strm.str();
+  }
+
+  // Char pointers
+  std::string printValue(const char *const val) {
+    std::ostringstream strm;
+    if (!val) {
+      strm << "<<<NULL>>>";
+    } else {
+      const char *cobj = val;
+      int i = 0;
+      strm << "\"";
+      while (*cobj != 0) {
+        strm << *cobj;
+        cobj++;
+        i++;
+      }
+      strm << "\"";
+    }
+    return strm.str();
+  }
+
+  std::string printValue(char *val) {
+    return printValue((const char *const) val);
+  }
+
+  // std::string
+  std::string printValue(const std::string & val) {
+    std::ostringstream strm;
+    strm << "\"";
+    strm << val;
+    strm << "\"";
+    return strm.str();
+  }
+
+  std::string printValue(const Value &value) {
+    std::ostringstream strm;
+
+    if (!value.isValid()) {
+      strm << "<<<invalid>>> @" << &value;
+    } else {
+      clang::ASTContext& C = value.getASTContext();
+      clang::QualType QT = value.getType();
+      strm << "boxes [";
+      strm << "("
+      << QT.getAsString(C.getPrintingPolicy())
+      << ") ";
+      clang::QualType valType = QT.getDesugaredType(C).getNonReferenceType();
+      if (C.hasSameType(valType, C.LongDoubleTy))
+        strm << value.simplisticCastAs<long double>() << "L";
```

Comment:
Can we invoke `printValue(long double)` here? Then we have that code (basically the 'L') only once.